traptor/traptor.py

```diff
@@ -593,11 +608,12 @@ def _main_loop(self):
 
                     if self.kafka_enabled == 'true':
                         try:
-                            self.kafka_conn.send(self.kafka_topic, enriched_data)
-                            self.logger.info("Tweet sent to kafka: {}".format(tweet.get('id_str', None)))
-                            dd_monitoring.increment('tweet_to_kafka_success')
+                            self.logger.info("Attempting to send tweet {} to kafka".format(tweet.get('id_str', None)))
+                            future = self.kafka_conn.send(self.kafka_topic, enriched_data)
+                            future.add_callback(self.kafka_success_callback, tweet)
```

Comment:
@joequery I think `tweet` here needs to be replaced with the enriched_data object which is the tweet with the additional Traptor fields.